File: schemas/email-campaign-marketing/drizzle/templates.ts
// templates: Reusable email templates with HTML and text bodies.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp, index } from "drizzle-orm/pg-core";
import { users } from "./users";

export const templates = pgTable(
  "templates",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    subject: text("subject"),
    htmlBody: text("html_body"),
    textBody: text("text_body"),
    fromName: text("from_name"),
    fromEmail: text("from_email"),
    createdBy: uuid("created_by").references(() => users.id, { onDelete: "set null" }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_templates_created_by").on(table.createdBy),
  ],
);

File: schemas/email-campaign-marketing/spacetimedb/templates.rs
// templates: Reusable email templates with subject, body, and sender defaults.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = templates, public)]
pub struct Template {
    #[primary_key]
    pub id: String, // UUID

    pub name: String,
    pub subject: Option<String>,
    pub html_body: Option<String>,
    pub text_body: Option<String>,
    pub from_name: Option<String>,
    pub from_email: Option<String>,

    #[index(btree)]
    pub created_by: Option<String>, // UUID, FK → users.id (set null)

    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
